clients/pkg/promtail/wal/encoding.go

```diff
@@ -0,0 +1,256 @@
+package wal
```

Comment:
should I add a comment on top of the copied files due to licensing, clarifying that?